### PR TITLE
Features/san 200 export all data

### PR DESCRIPTION
**Pull request by JanKanis**
coredb  changes to support export of hd data

### CODE REVIEW

**Comment by cataphract** [comment]
Replace `a4f9903` with proper message and adhere to the 50/72 rule (first line no more than 50 chars -- if possible --, followed by an empty line and then text wrapped at col 72, except when it can't be wrapped, e.g. for URLs), which was grossly violated on `1c9f11b`.

**Comment by cataphract** [comment]
Pending resolution of https://github.com/thehyve/transmart-core-api/pull/1

File: src/groovy/org/transmartproject/db/dataquery/CollectingTabularResult.groovy [hunk 1]
```diff
@@ -149,7 +149,7 @@ abstract class CollectingTabularResult<C, R extends DataRow>
         }
 
         results.next()
-        R row = nextRow
+        R row = getNextRow()
```
Comment (haarts) [inline src/groovy/org/transmartproject/db/dataquery/CollectingTabularResult.groovy:152]: Why is this change necessary?

File: src/groovy/org/transmartproject/db/dataquery/highdim/AbstractDataRow.groovy [hunk 2]
```diff
@@ -23,4 +23,9 @@ abstract class AbstractDataRow implements DataRow<AssayColumn, Object> {
     Iterator<Object> iterator() {
         data.iterator()
     }
+
+    Map<String, Object> getAssociatedData() {
+        [label: label]
+    }
+
```
Comment (haarts) [inline src/groovy/org/transmartproject/db/dataquery/highdim/AbstractDataRow.groovy]: :fish:

File: src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy [hunk 3]
```diff
@@ -28,6 +30,12 @@ class AcghModule extends AbstractHighDimensionDataTypeModule {
 
     final String name = 'acgh'
 
+    final ImmutableSet<String> dataProperties = ImmutableSet.copyOf('trialName chipCopyNumberValue ' +
+            'segmentCopyNumberValue flag probabilityOfLoss probabilityOfNormal probabilityOfGain ' +
+            'probabilityOfAmplification'.split())
```
Comment (haarts) [inline src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy]: What IS it? I don't get what's going on here.

Comment (haarts) [inline src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy]: Actually I think I do and I think it's a bad idea. You are splitting a string because you are too lazy to put quotes around the substrings.

Comment (cataphract) [inline src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy]: I agree. `['foo', 'bar']` is much clearer.

File: src/groovy/org/transmartproject/db/dataquery/highdim/mrna/MrnaModule.groovy [hunk 4]
```diff
@@ -138,7 +146,8 @@ class MrnaModule extends AbstractHighDimensionDataTypeModule {
 
             if (collected.size() > 1) {
                 /* modify 1st element with info from subsequents */
-                collected[0].geneSymbol = collected*.geneSymbol.join('/')
+                collected[0].geneSymbol = collected*.geneSymbol.grep().join('/') ?: null
+                collected[0].geneId = collected*.geneId.grep().join('/') ?: null
```
Comment (cataphract) [inline src/groovy/org/transmartproject/db/dataquery/highdim/mrna/MrnaModule.groovy]: Why? The commit messages don't mention this.

File: test/integration/org/transmartproject/db/dataquery/highdim/mrna/MrnaGeneDataConstraintTests.groovy [hunk 5]
```diff
@@ -67,11 +67,11 @@ class MrnaGeneDataConstraintTests {
...
 
         /* keywords ids for genes BOGUSCPO, BOGUSRQCD1 */
-        def testee = mrnaModule.createDataConstraint(
-                DataConstraint.SEARCH_KEYWORD_IDS_CONSTRAINT,
-                [ keyword_ids: testData.searchKeywords.findAll {
-                    [ 'BOGUSCPO', 'BOGUSRQCD1' ].contains(it.keyword)
-                }*.id])
+        def testee = mrnaModule.createDataConstraint([keyword_ids: testData.searchKeywords.findAll {
+            ['BOGUSCPO', 'BOGUSRQCD1'].contains(it.keyword)
+        }*.id],
```
Comment (cataphract) [inline test/integration/org/transmartproject/db/dataquery/highdim/mrna/MrnaGeneDataConstraintTests.groovy:72]: The line indentation is incorrect here.

File: src/groovy/org/transmartproject/db/dataquery/CollectingTabularResult.groovy (hunk 1 above)
Comment (JanKanis) [inline src/groovy/org/transmartproject/db/dataquery/CollectingTabularResult.groovy:152]: nextRow doesn't behave as a property, it changes state and returns a different thing the next time it's called, so this is clearer. A few lines down it was already called as getNextRow().

File: src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy (hunk 3 above)
Comment (JanKanis) [inline src/groovy/org/transmartproject/db/dataquery/highdim/acgh/AcghModule.groovy]: I find 'string'.split() easier to work with if there are lots of symbols, but I'll change it.

File: src/groovy/org/transmartproject/db/dataquery/highdim/mrna/MrnaModule.groovy (hunk 4 above)
Comment (JanKanis) [inline src/groovy/org/transmartproject/db/dataquery/highdim/mrna/MrnaModule.groovy]: If there's a null, it results in the string "foo/null", that doesn't seem very useful.

File: test/integration/org/transmartproject/db/dataquery/highdim/mrna/MrnaGeneDataConstraintTests.groovy (hunk 5 above)
Comment (JanKanis) [inline test/integration/org/transmartproject/db/dataquery/highdim/mrna/MrnaGeneDataConstraintTests.groovy:72]: blame intellij :)

Comment (haarts) [inline test/integration/org/transmartproject/db/dataquery/highdim/mrna/MrnaGeneDataConstraintTests.groovy:72]: Blame the gun when you shoot someone? ;)
